x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/types.ts

```diff
@@ -16,43 +16,73 @@ export interface Mappings {
   indexes: Record<string, IndicesGetMappingIndexMappingRecord>;
 }
 
-export interface AllowedValue {
-  description?: string;
-  expected_event_types?: string[];
-  name?: string;
-}
+export interface EcsFieldMetadata {
+  dashed_name: string;
+  description: string;
+  flat_name: string;
+  level: string;
+  name: string;
+  normalize: string[];
+  short: string;
+  type: string;
 
-export interface EcsMetadata {
   allowed_values?: AllowedValue[];
-  dashed_name?: string;
-  description?: string;
-  example?: string;
-  flat_name?: string;
+  beta?: string;
+  doc_values?: boolean;
+  example?: string | number | boolean;
+  expected_values?: string[];
   format?: string;
   ignore_above?: number;
-  level?: string;
-  name?: string;
-  normalize?: string[];
+  index?: boolean;
+  input_format?: string;
+  multi_fields?: MultiField[];
+  object_type?: string;
+  original_fieldset?: string;
+  output_format?: string;
+  output_precision?: number;
+  pattern?: string;
   required?: boolean;
-  short?: string;
-  type?: string;
+  scaling_factor?: number;
+}
+
+export interface AllowedValue {
+  description: string;
+  name: string;
+  expected_event_types?: string[];
+  beta?: string;
 }
 
-export type EnrichedFieldMetadata = EcsMetadata & {
-  hasEcsMetadata: boolean;
+export interface MultiField {
+  flat_name: string;
+  name: string;
+  type: string;
+}
```

Comment:
For reviewers:

This is a precise cumulative type inference of a given field from `@elastic/ecs` npm package. All required fields are actually required in `ecs_flat.js` file that is imported and used in our code.